execution/scan/matrix_selector.go

```diff
@@ -256,13 +256,21 @@ loop:
 		switch buf.Next() {
 		case chunkenc.ValNone:
 			break loop
-		case chunkenc.ValHistogram, chunkenc.ValFloatHistogram:
-			t, h := buf.AtFloatHistogram()
+		case chunkenc.ValHistogram:
+			t, h := buf.AtHistogram()
```

Comment:
We need the same fix here: https://github.com/thanos-community/promql-engine/blob/main/execution/scan/matrix_selector.go#L281

Reply:
This should

> Looks good, just a few comments from my side.

I addressed the comments. Can you ptal.